pkg/asset/machines/azure/machinesets.go

```diff
@@ -22,61 +22,119 @@ func MachineSets(clusterID string, config *types.InstallConfig, pool *types.Mach
 	}
 	platform := config.Platform.Azure
 	mpool := pool.Platform.Azure
+	azs := mpool.Zones
 
-	total := int32(0)
+	total := int64(0)
 	if pool.Replicas != nil {
-		total = int32(*pool.Replicas)
+		total = *pool.Replicas
 	}
-
+	numOfAZs := int64(len(azs))
 	var machinesets []*clusterapi.MachineSet
-	provider, err := provider(platform, mpool, osImage, userDataSecret, clusterID, role)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to create provider")
-	}
-	// TODO(serbrech): Implement AZ support
-	//name := fmt.Sprintf("%s-%s-%s", clustername, pool.Name, az)
-	name := fmt.Sprintf("%s-%s", clusterID, pool.Name)
-	mset := &clusterapi.MachineSet{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: "machine.openshift.io/v1beta1",
-			Kind:       "MachineSet",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: "openshift-machine-api",
-			Name:      name,
-			Labels: map[string]string{
-				"machine.openshift.io/cluster-api-cluster":      clusterID,
-				"machine.openshift.io/cluster-api-machine-role": role,
-				"machine.openshift.io/cluster-api-machine-type": role,
+	for idx, az := range azs {
+		replicas := int32(total / numOfAZs)
+		if int64(idx) < total%numOfAZs {
+			replicas++
+		}
+		provider, err := provider(platform, mpool, osImage, userDataSecret, clusterID, role, &idx)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create provider")
+		}
+		name := fmt.Sprintf("%s-%s-%s", clusterID, pool.Name, az)
```

Comment:
since the AZs in Azure are number: `1,2,3` it might be useful to keep names as `<id>-<pool>-<region><zone>`